concent_api/signing_service_integration_test_recovery.py

```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/env python3
+
+from configparser import ConfigParser
+from time import sleep
+from typing import Any
+import os
+import socket
+import subprocess
+import sys
+
+from api_testing_common import assert_condition
+from api_testing_common import count_fails
+from common.helpers import get_current_utc_timestamp
+from middleman_protocol.message import GolemMessageFrame
+from middleman_protocol.concent_golem_messages.message import SignedTransaction
+from middleman_protocol.stream import send_over_stream
+from middleman_protocol.stream import unescape_stream
+from signing_service_testing_common import Components
+from signing_service_testing_common import create_golem_message_frame_with_correct_transaction_signing_request
+from signing_service_testing_common import run_tests
+from signing_service_testing_common import send_message_to_middleman_and_receive_response
+from signing_service_testing_common import send_message_to_middleman_without_response
+
+
+REQUIRED_COMPONENTS = []  # type: ignore
+
+SLEEP_TIME_AFTER_SPAWNING_OR_KILLING_PROCESS = 10
+
+os.environ.setdefault("DJANGO_SETTINGS_MODULE", "concent_api.settings")
+
+
+def run_middleman() -> subprocess.Popen:
+    return subprocess.Popen(
+        [
+            'python',
+            'manage.py',
+            'middleman'
+        ]
+    )
+
+
+def run_signing_service() -> subprocess.Popen:
+    return subprocess.Popen(
+        [
+            'python',
+            '-m',
+            'signing_service.signing_service',
+            '127.0.0.1',
+            '"85cZzVjahnRpUBwm0zlNnqTdYom1LF1P1WNShLg17cmhN2UssnPrCjHKTi5susO3wrr/q07eswumbL82b4HgOw=="',
+            '--signing-service-private-key',
+            '"GfkWzCRGL/Me2wk41mS0NRSch2XMpgIXYLRaasmcozw="',
+            '--ethereum-private-key',
+            '"M2ExMDc2YmY0NWFiODc3MTJhZDY0Y2NiM2IxMDIxNzczN2Y3ZmFhY2JmMjg3MmU4OGZkZDlhNTM3ZDhmZTI2Ng=="'
+        ],
+        cwd='../signing_service/'
+    )
+
+
+@count_fails
+def test_case_0_prove_of_concept(config: ConfigParser, **kwargs: Any) -> None:
+    """
+    1. Spawn MiddleMan and SigningService processes.
+    2. Client sends GolemMessageFrame with correct TransactionSigningRequest to MiddleMan
+       and receives response to prove that connection works.
+    """
+
+    try:
+        middleman_process = run_middleman()
+        signing_service_process = run_signing_service()
+
+        # Waiting for MiddleMan and SigningService to start.
+        sleep(SLEEP_TIME_AFTER_SPAWNING_OR_KILLING_PROCESS)
```

Comment:
I don't know if it is possible but can you instead of waiting check (e.g. in loop with short sleep time) if MM and SS started and break the loop when they are? It will be much faster

Reply:
No, it is not possible, you can't get info directly from the process, only get return code or parse `stdout` and `stderr` but for test purposes it would be overkill.

Reply:
ok, thanks